### ISSUE TITLE
No alignment after 800 epochs when training from scratch / cold-start

### ISSUE THREAD

**Issue by pmunaretto** [NVIDIA/tacotron2#480]
Hi all,

I'm currently training a model based on a corpus of the M-AILABS German language dataset that is trimmed with Yeongtae's preprocessing script (~18 hours total). When I warm-start from the pretrained English model, I can see first signs of alignment after 1k iterations and almost perfect alignment after 7k iterations. The samples also sound quite good already.

Now I tried to train Tacotron2 from scratch to achieve better results as mentioned in https://github.com/NVIDIA/tacotron2/issues/373#issuecomment-644672311. 

However, I had no success in doing so. After 156.000 iterations (876 epochs), I do not see any alignment.

<img src="https://user-images.githubusercontent.com/24635295/116461538-a6985b80-a868-11eb-8299-d0d22e7a1ac0.gif" width="400">

<img src="https://user-images.githubusercontent.com/24635295/116461871-11e22d80-a869-11eb-8a42-abf61d0c6dde.png" width="200">

<img src="https://user-images.githubusercontent.com/24635295/116461894-1ad2ff00-a869-11eb-8c61-5adbf3884ea1.png" width="200">

Im training on default parameters with FP16 enabled and a batch_size of 56. I reduced the learning rate on epoch 500 according to NVIDIAs DeepLearningExamples repo where they use the following scheme:

- Epoch 0-500: 1e-3
- Epoch 500-1000: 1e-4
- Epoch 1000-1500: 1e-5

The pretrained FP32 model from NVIDIA is only learned for 1.500 epochs and I'm getting closer to this without seeing any progress so far (besides the reduced loss). Should I stop training at this point? Any help is appreciated. Thanks!

Edit: I synthesized a few sentences on checkpoint_167000. As expected, there is no alignment. The synthesized samples sound like some "gibberish" where parts of the original wavs are mixed together randomly.

**Comment by mmehiretu2019** [comment]
I'm running into this same issue after 390 epochs for a custom dataset. @G0x0pher did you figure out what the issue is?

**Comment by pmunaretto** [comment]
> I'm running into this same issue after 390 epochs for a custom dataset. @G0x0pher did you figure out what the issue is?

Unfortunately no. I sticked to warmstarting on the pretrained model.

**Comment by mmehiretu2019** [comment]
> > I'm running into this same issue after 390 epochs for a custom dataset. @G0x0pher did you figure out what the issue is?
> 
> Unfortunately no. I sticked to warmstarting on the pretrained model.

I see. After some investigation, I'm suspecting my issue could be related to aggressive trimming of my audio files. My original dataset is fairly clean, and had very little silence at the end of each audio. However, I went ahead and trimmed all audio files using librosa with max db=42 and that removed about an hour and half from the total duration of the dataset. Total duration of dataset before trimming was 18:35:00 and after trimming it was 17:00:00. Knowing the original dataset didn't have that much silence, trimming might have removed non-silence part of the audio too. 

Anyways, I have now restarted training from the non-trimmed dataset. I will let you know how it goes
